src/View/Kitchen/Kitchen.ts

```diff
@@ -1,66 +1,91 @@
-import Drink, { DrinkMap } from '@/Model/Drink';
-import { OrderControllerObsever } from '@/Controller/OrderController';
+import View from '@/core/View';
+import type Order from '@/Model/Order';
+import Drink from '@/Model/Drink';
+import { ORDER_STORE } from '@/Stores/orderStore/constants';
 
-class Kitchen implements OrderControllerObsever {
-  kitchenControllerElement = document.getElementById('coffee-list-kitchen')!;
+class Kitchen extends View {
+  $kitchenController = document.getElementById('coffee-list-kitchen')!;
+  $kitchenCover = document.getElementById('kitchen-cover')!;
   buttons: { [id: string]: HTMLButtonElement } = {};
-  kitchenCoverElement = document.getElementById('kitchen-cover')!;
   isKichenOpen = false;
 
   constructor() {
-    const children = Array.from(this.kitchenControllerElement.children);
-    this.findCoffeeCategoryButton(children);
-    this.event();
+    super();
+    const children = Array.from(this.$kitchenController.children);
+    this.mapCoffeeCategoryButton(children);
   }
 
-  private event = () => {
-    this.kitchenCoverElement.addEventListener('click', () => {
+  protected bindEvents = () => {
+    this.$kitchenCover.addEventListener('click', () => {
       alert('주문해주세요!');
     });
+
+    window.addEventListener(ORDER_STORE.event, e => {
+      const { type, store, payload } = e.detail;
+      if (type === ORDER_STORE.types.ADD) {
+        const addedOrder = payload as Order;
+
+        return this.addDrink(addedOrder.drink);
+      }
+
+      if (type === ORDER_STORE.types.DELETE) {
+        const storedOrders = store.orders as Order[];
+        const deletedOrder = payload as Order;
+        const isSameCoffeeExist = storedOrders.find(order => order.drink.name === deletedOrder.drink.name);
```

Comment:
```suggestion
        const isSameCoffeeExist = storedOrders.isSameCoffeeExist(deleteOrder);
```
`storedOrders` 에게 메세지를 전달해서 직접 확인 하도록 해보는건 어때요~?

Reply:
setOrders가 단순 Array 객체이기 때문에, 별도의 method를 주기가 어려워 
drink 객체에 isSameDrink(otherDrink: Drink) 라는 메소드를 새로 만들어줬습니다.

이런 방법은 어떤가요?

Reply:
메소드로 변경하는 것도 좋네요! 😊
관심이 있으시다면 `storeOrders`를 `일급컬렉션` 으로 바꿔보는 것 도 참고 해 보실 수 있을 것 같아요!

참조 : [일급 컬렉션](https://velog.io/@seongwon97/%EC%9D%BC%EA%B8%89-%EC%BB%AC%EB%A0%89%EC%85%98%EC%9D%B4%EB%9E%80)